src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpCollectionExpressionRewriter.cs

```diff
@@ -588,5 +588,36 @@ static IEnumerable<SyntaxNode> GetElementComponents(StatementSyntax statement)
 
         static StatementSyntax UnwrapEmbeddedStatement(StatementSyntax statement)
             => statement is BlockSyntax { Statements: [var innerStatement] } ? innerStatement : statement;
+
+        static ExpressionSyntax ConvertExpression(
```

Comment:
this moved from common code to be specific to collection-expressions and specific to colleciton-initializers.  it turned out sharing just was not possible.  Especially because if you had `x[a] = b` then collection expressions would add this as just `b` whereas collection initializers would add it as `[a] = b`.  

Given that, it would be too many contortions to try to share. so i just pulled out for eahc.